File: spring-cloud-integrated-demo/order-service/src/main/java/com/github/jccode/springcloud/integrateddemo/order/form/OrderForm.java
package com.github.jccode.springcloud.integrateddemo.order.form;

import javax.validation.Valid;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;
import java.util.List;

public class OrderForm {

    @NotNull
    @Min(1)
    private Integer userId;

    private String address;

    @Valid
    private List<OrderItemForm> items;

    public Integer getUserId() {
        return userId;
    }

    public void setUserId(Integer userId) {
        this.userId = userId;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public List<OrderItemForm> getItems() {
        return items;
    }

    public void setItems(List<OrderItemForm> items) {
        this.items = items;
    }
}

File: spring-cloud-integrated-demo/order-service/src/main/scala/com/github/jccode/springcloud/integrateddemo/order/form/OrderForm.scala
package com.github.jccode.springcloud.integrateddemo.order.form

import javax.validation.Valid
import javax.validation.constraints.{Min, NotNull}

import scala.beans.BeanProperty

class OrderForm {

  // 1.得用这种方式,不能用case class方式;
  // 2.而且primary constructor必须是无参的.要不然 Validation 不起作用.
  @BeanProperty @NotNull @Min(1) var userId: Integer = _
  @BeanProperty var address: String = _
  @BeanProperty @Valid @NotNull var items: java.util.List[OrderItemForm] = _

  override def toString = s"OrderForm($userId, $address, $items)"
}
